core/src/main/java/io/grpc/internal/AbstractServerImplBuilder.java

```diff
@@ -175,11 +176,15 @@ public ServerImpl build() {
     StatsContextFactory statsFactory =
         this.statsFactory != null ? this.statsFactory : Stats.getStatsContextFactory();
     if (statsFactory != null) {
-      CensusStreamTracerModule census =
-          new CensusStreamTracerModule(statsFactory, GrpcUtil.STOPWATCH_SUPPLIER);
-      tracerFactories.add(census.getServerTracerFactory());
+      CensusStatsModule censusStats =
+          new CensusStatsModule(statsFactory, GrpcUtil.STOPWATCH_SUPPLIER);
+      tracerFactories.add(censusStats.getServerTracerFactory());
     }
+    CensusTracingModule censusTracing =
+        new CensusTracingModule(Tracing.getTracer(), Tracing.getBinaryPropagationHandler());
```

Comment:
Per offline discussion, in order to prevent premature wire-format changes from breaking existing users, we should keep both stats and tracing flag-protected until their wire format stabilized. I have added two static booleans to `GrpcUtil`.